core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +560,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): Option[FsApplicationAttemptInfo] = {
```

Comment:
@srowen while you're right the return value could be empty if the filter specified filtered out the app start/end events, but then I also see that ideally the way to parse out data from replayed events is for callers to set appropriate listeners of their choice on the ReplayListenerBus instance specified for event replay. And the listeners capture event data etc. This particular replay () method in FsHistoryProvider looks like a helper for the two call sites and not much more, so maybe not so much of a problem?